tests/playbooks/tests_states.yml

```diff
@@ -4,74 +4,134 @@
-            state: down
-    - include_tasks: tasks/assert_device_absent.yml
-    - include_tasks: tasks/assert_profile_absent.yml
+    - block:
+        - include_tasks: tasks/run_test.yml
+          vars:
+            lsr_description: I can create a profile without autoconnect
+            lsr_setup:
+              - tasks/delete_interface.yml
+              - tasks/assert_device_absent.yml
+            lsr_test:
+              - tasks/create_bridge_profile_no_autoconnect.yml
+            lsr_assert:
+              # Device should be absent because of autoconnect: false
+              - tasks/assert_device_absent.yml
+              - tasks/assert_profile_present.yml
+            lsr_cleanup:
+              - tasks/cleanup_profile+device.yml
+      tags:
+        - tests::states:create_without_autoconnect
 
-    # test case (downing a non existent profile)
-    # I can set down and remove a non existent profile
-    # This will be rescued until issue 188 is fixed
-    - name: Verify that the play fails with the expected error message
-      block:
-        - name: Set down non existent profile
-          include_role:
-            name: linux-system-roles.network
+    - block:
+        - include_tasks: tasks/run_test.yml
           vars:
-            network_connections:
-              - name: non_existent
-                state: down
-                persistent_state: absent
+            lsr_description: I can activate an existing profile
+            lsr_setup:
+              - tasks/create_bridge_profile.yml
+            lsr_test:
+              - tasks/activate_profile.yml
+            lsr_assert:
+              - tasks/assert_device_present.yml
+              - tasks/assert_profile_present.yml
+            lsr_cleanup:
+              - tasks/cleanup_profile+device.yml
+      tags:
+        - tests::states:activate
 
-        - name: 'UNREACH'
-          set_fact:
-            __network_test_failed: false
+    - block:
+        - include_tasks: tasks/run_test.yml
+          vars:
+            lsr_description: I can remove an existing profile without taking it
+              down
+            lsr_setup:
+              - tasks/create_bridge_profile.yml
+              - tasks/activate_profile.yml
+            lsr_test:
+              - tasks/remove_profile.yml
+            lsr_assert:
+              - tasks/assert_device_present.yml
+              - tasks/assert_profile_absent.yml
+            lsr_cleanup:
+              - tasks/cleanup_profile+device.yml
+      tags:
+        - tests::states:remove_up
 
-      rescue:
-        - name: Role has failed when it should have
-          set_fact:
-            __network_test_failed: true
+    - block:
+        - include_tasks: tasks/run_test.yml
+          vars:
+            lsr_description: I can take a profile down that is absent
+            lsr_setup:
+              - tasks/create_bridge_profile.yml
+              - tasks/activate_profile.yml
+              - tasks/remove_profile.yml
```

Comment:
I think that here, too, you can remove the `lsr_setup` above and `lsr_cleanup` in the previous block to chain the tests together.